File: src/backend/studio-reservation/src/main/java/org/taonaw/studio_reservation/domain/model/usageFeeSetting/ApplicablePeriod.java
package org.taonaw.studio_reservation.domain.model.usageFeeSetting;

import lombok.EqualsAndHashCode;
import lombok.Getter;
import org.taonaw.studio_reservation.domain.model.shared.DateRange;

import java.time.LocalDate;

@Getter
@EqualsAndHashCode(callSuper = true)
public class ApplicablePeriod extends DateRange {
    public ApplicablePeriod(LocalDate startDate, LocalDate endDate) {
        super(startDate, endDate);
    }
}

File: src/backend/facility/src/main/kotlin/org/taonaw/facility/domain/model/shared/ApplicablePeriod.kt
package org.taonaw.facility.domain.model.shared

import java.time.LocalDate

data class ApplicablePeriod(
        val start: LocalDate,
        val end: LocalDate) {

    init {
        require(start < end) { "終了日時を開始日時より後でなければなりません。" }
    }

    fun isOverlapping(other: ApplicablePeriod): Boolean {
        return start <= other.end && other.start <= end
    }
}